benchmark/pytest/analysis/test_optimization_group.py

```diff
@@ -44,7 +44,6 @@ class BenchmarkMegacomplex(Megacomplex):
     def calculate_matrix(
         self,
         dataset_model,
-        global_index: int | None,
```

Comment:
Note for myself: this is not enough

Reply:
Ok, if you wish to resolve it another way in this PR, then make the commit to do so, or create a new issues to do it later and mark this comment as resolved.

Reply:
Please address in the subsequent PRs.